sftk/video_handler.py

```diff
@@ -0,0 +1,981 @@
+        _skip_init_logs: bool = False,
+    ):
+        self.s3_handler = s3_handler
+        self.prefix = prefix.strip()
+        self.gopro_prefix = gopro_prefix
+        self.delete_originals = delete_originals
+        self.test_mode = test_mode
+        self.download_threads = download_threads 
+        self.parallel_drops = parallel_drops
+        self.sequential_download = sequential_download
+        self._skip_init_logs = _skip_init_logs  # Store the flag
+        
+        # Suppress S3Handler logs in worker processes
+        if _skip_init_logs:
+            logging.getLogger('sftk.s3_handler').setLevel(logging.WARNING)
+
+        if not _skip_init_logs:
+            logger.info(f"{'='*80}")
+            logger.info(f"🎥 VideoProcessor Initialization")
+            logger.info(f"{'='*80}")
+            logger.info(f"   Prefix: '{prefix}'")
+            logger.info(f"   GoPro prefix: '{gopro_prefix}'")
+            logger.info(f"   Test mode: {test_mode}")
+            logger.info(f"   Parallel drops: {parallel_drops}")
+            logger.info(f"   Download threads: {download_threads}")
+            logger.info(f"   Sequential download: {sequential_download}")
+            logger.info(f"   Delete originals: {delete_originals}")
+            logger.info(f"{'='*80}")
+        
+        self.download_dir = Path("downloaded_movies")
+        self.output_dir = Path("concatenated_videos")
+
+        self.download_dir.mkdir(exist_ok=True)
+        self.output_dir.mkdir(exist_ok=True)
+
+        self.ffmpeg_path = _find_ffmpeg()
+        if not self.ffmpeg_path:
+            raise RuntimeError(
+                "ffmpeg not found. Please install ffmpeg and ensure it's in your system's PATH."
+            )
+        else:
+            logger.info(f"✓ Found ffmpeg at: {self.ffmpeg_path}")
+
+        # Pre-fetch and filter movies to provide upfront information
+        logger.info(f"🔍 Fetching movie list from S3 with prefix: '{self.prefix}'")
+        self.movies_df = self.get_movies_df(prefix=self.prefix)
+        logger.info(f"📁 Found {len(self.movies_df):,} movie files with the specified prefix")
+        
+        movies_df_with_parts = _add_path_parts_to_df(self.movies_df)
+        self.filtered_df = get_filtered_movies_df(
+            movies_df=movies_df_with_parts.copy(), 
+            gopro_prefix=self.gopro_prefix
+        )
+        
+        num_drops = self.filtered_df['DropID'].nunique() if not self.filtered_df.empty else 0
+        logger.info(f"🎬 Found {num_drops} drop(s) that require concatenation")
+        logger.info(f"{'='*80}\n")
+
+    def process_gopro_videos(self) -> None:
+        """Process GoPro videos by DropID."""
+        if self.filtered_df.empty:
+            logger.info("ℹ️  No GoPro videos found that require concatenation")
+            return
+
+        # CRITICAL: Use drop_duplicates on DropID to ensure each drop appears only once
+        unique_drops_df = self.filtered_df.drop_duplicates(subset=['DropID'])
```

Comment:
I imagine the duplicate DropID issue should be fixed before/during validation period, meaning before getting into the aws bucket?